File: app/src/main/java/com/u1fukui/bbs/model/BbsThread.java
package com.u1fukui.bbs.model;

import java.io.Serializable;
import java.util.Date;

import lombok.AllArgsConstructor;

@AllArgsConstructor
public class BbsThread implements Serializable {

    private static final long serialVersionUID = -4762213969235749270L;

    public long id;

    public String title;

    public User author;

    public int commentCount;

    public Date createdAt;

    public Date updatedAt;
}

File: app/src/main/java/com/u1fukui/bbs/model/BbsThread.kt
package com.u1fukui.bbs.model

import se.ansman.kotshi.JsonSerializable
import java.io.Serializable
import java.util.Date

@JsonSerializable
data class BbsThread(
    val id: Long,
    val title: String,
    val author: User,
    val commentCount: Int,
    val createdAt: Date,
    val updatedAt: Date
) : Serializable {

    companion object {

        private const val serialVersionUID = -4762213969235749270L
    }
}
